### PR TITLE
[Java] AE: Add JavaDocs

### PR DESCRIPTION
**Pull request by sushantmimani**
- Add JavaDocs for all public methods
- Enforce JavaDoc check during the build phase
- Bump version to `0.1.3-alpha` for explicit no-op release

### CODE REVIEW

File: java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SessionFactory.java [hunk 1]
```diff
@@ -44,6 +44,15 @@
   private final KeyManagementService keyManagementService;
   private final Cache<String, CachedSession> sessionCache;
 
+  /**
+   * Constructor for SessionFactory.
+   * @param productId The product Id.
+   * @param serviceId The service Id.
+   * @param metastore A {@link Metastore} object.
+   * @param systemKeyCache Cache for SystemKeys.
```
Comment (nikoo28) [inline java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SessionFactory.java:52]: should be `system keys`. Also check capitalization. The first word after the param variable should not be capitalized.

File: java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SessionFactory.java [hunk 2]
```diff
@@ -167,24 +176,44 @@ void releaseShared(final String partitionId) {
     });
   }
 
+  /**
+   * Uses the {@code partitionId} to get the implementation of {@code EnvelopeEncryption}.
+   * @param partitionId The partition Id.
+   * @return a session that encrypts a json payload and stored as byte[].
```
Comment (nikoo28) [inline java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SessionFactory.java:182]: the first word after `return` should be capitalized.

File: java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SessionFactory.java [hunk 3]
```diff
@@ -167,24 +176,44 @@ void releaseShared(final String partitionId) {
...
   public Session<JSONObject, byte[]> getSessionJson(final String partitionId) {
     EnvelopeEncryption<byte[]> envelopeEncryption = getEnvelopeEncryptionBytes(partitionId);
 
     return new SessionJsonImpl<>(envelopeEncryption);
   }
 
+  /**
+   * Uses the {@code partitionId} to get the implementation of {@code EnvelopeEncryption}.
+   * @param partitionId The partition Id.
+   * @return a session that encrypts a byte[] payload and stored as byte[].
```
Comment (nikoo28) [inline java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/SessionFactory.java:193]:
```suggestion
   * @return A session that encrypts a byte[] payload and stored as byte[].
```

File: java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/DefaultPartition.java [hunk 4]
```diff
@@ -2,6 +2,12 @@
 
 public class DefaultPartition extends Partition {
 
+  /**
+   * Constructor for {@link DefaultPartition}.
```
[the remark this answers is not in the document]
Comment (sushantmimani) [reply java/app-encryption/src/main/java/com/godaddy/asherah/appencryption/DefaultPartition.java:6]: `link` is _also_ used for enum, but not specifically for that.
https://docs.oracle.com/en/java/javase/11/docs/specs/doc-comment-spec.html#link

**Comment by sushantmimani** [comment]
Will break this in smaller, package level, PRs for easier reviews.

**Pull request closed without merging**